middleware/prometheus.go

```diff
@@ -40,6 +41,14 @@ func InstrumentRoundTripper(roundTripper http.RoundTripper, service string) http
 		Help:        "A gauge of in-flight requests for the wrapped client.",
 		ConstLabels: prometheus.Labels{"service": service},
 	})
+	// attempt to register, if already registered use the registered one
+	if err := prometheus.Register(inFlightGauge); err != nil {
+		var are *prometheus.AlreadyRegisteredError
+		if errors.As(err, &are) {
+			// already registered.
+			inFlightGauge = are.ExistingCollector.(prometheus.Gauge)
```

Comment:
Should this be a `Gauge` or a `GaugeVec`

Reply:
Gauge is an interface that extends off of Collector.  Anything that implements Gauge also implements Collector